lib/Controller/FolderApiV2Controller.php

```diff
@@ -0,0 +1,120 @@
+<?php
+/**
+ * Nextcloud - News
+ *
+ * This file is licensed under the Affero General Public License version 3 or
+ * later. See the COPYING file.
+ *
+ * @author    Paul Tirk <paultirk@paultirk.com>
+ * @copyright 2020 Paul Tirk
+ */
+
+namespace OCA\News\Controller;
+
+use \OCP\IRequest;
+use \OCP\IUserSession;
+use \OCP\AppFramework\Http;
+
+use \OCA\News\Service\FolderServiceV2;
+use \OCA\News\Service\ItemServiceV2;
+use \OCA\News\Service\Exceptions\ServiceNotFoundException;
+
+class FolderApiV2Controller extends ApiController
+{
+    use ApiPayloadTrait;
+    use JSONHttpErrorTrait;
+
+    /**
+     * @var FolderServiceV2
+     */
+    private $folderService;
+
+    /**
+     * @var ItemServiceV2
+     */
+    private $itemService;
+
+    public function __construct(
+        IRequest $request,
+        IUserSession $userSession,
+        FolderServiceV2 $folderService,
+        ItemServiceV2 $itemService
+    ) {
+        parent::__construct($request, $userSession);
+
+        $this->folderService = $folderService;
+        $this->itemService = $itemService;
+    }
+
+    /**
+     * @NoAdminRequired
+     * @NoCSRFRequired
+     * @CORS
+     *
+     * @param string $name
+     * @return array|mixed|\OCP\AppFramework\Http\JSONResponse
+     */
+    public function create(string $name)
+    {
+        if (empty($name)) {
```

Comment:
It can't be null because of the typehint so all you want to check for is an empty string
```suggestion
        if (trim($name) === '') {
```